code/modules/research/stock_parts.dm

```diff
@@ -56,12 +56,16 @@ If you create T5+ please take a pass at mech_fabricator.dm. The parts being good
 
 	RegisterSignal(src, COMSIG_ATOM_ENTERED, .proc/on_part_entered)
 
-/obj/item/storage/part_replacer/bluespace/proc/on_part_entered(datum/source, obj/item/I)
+/obj/item/storage/part_replacer/bluespace/proc/on_part_entered(datum/source, obj/item/inserted_component)
 	SIGNAL_HANDLER
-	if(!istype(I, /obj/item/stock_parts/cell))
+	if(inserted_component.reagents && length(inserted_component.reagents.reagent_list))
+		inserted_component.reagents.clear_reagents()
+		to_chat(usr, span_notice("\The [src] churns as \the [inserted_component] has its reagents emptied into bluespace."))
```

Comment:
`usr`? 😕 

Can we get a proper `user` passed into the signal, or is that too gross?

Reply:
That signal isn't necessarily one where you'd always expect a user.

I wouldn't sat a change like that is entirely out of scope, but it would mean updating every signal handler for this signal to include the new arg. What are your thoughts?

Reply:
Ah yeah, COMSIG_ATOM_ENTERED is quite a low level one, so it would be pretty gross to change all of them.

What about the storage component, is the hook's more natural place there?

Reply:
> 
> 
> Ah yeah, COMSIG_ATOM_ENTERED is quite a low level one, so it would be pretty gross to change all of them.
> 
> What about the storage component, is the hook's more natural place there?

Storage component doesn't have any signal logic for "something was inserted into a storage component" and handles item insertion signals via forceMove which relies on `/atom/Entered`/COMSIG_ATOM_ENTERED

It seems a bit redundant to add a signal for storage that is identical to COMSIG_ATOM_ENTERED.

Dunno though. I'd be happy to gather other maintainer feedback on this too. I have no aversion to using usr because I work heavily in tgui. If a dedicated `mob/user` has been passed down the call stack I'll use it, but stuff like tgui's ui_act leverage `usr` heavily too (when they could probably just as easily pass it down the call stack)

Reply:
Could you talk about why `usr` is bad, and why we should be using `user` instead? I'm trying to work out where this case lies on practicality vs purity.

Reply:
do you need the \the? I always forget

Reply:
The \the is because I don't want to have a `The the chem dispenser` or `the the bluespace beaker`. It would that be \improper? I can't remember?

Reply:
You very rarely need `\the` as BYOND makes it implicit.

Reply:
I agree that `usr` is crap though, and we should avoid it--I think even the reference says this.

Reply:
it should autopopulate the `the` when you use [src] or equivalent (not [name], must be the reference).

"[src] is cute" should translate into "The girl is cute" iirc.